ciso/_ciso.pyx

```diff
@@ -40,14 +40,15 @@ def zslice(double[:, :, ::1] q,
 
     cdef double[:, ::1] q_iso = np.empty((M, L), dtype=np.float64)
 
-    for i in range(L):
-        for j in range(M):
-            q_iso[j, i] = mask_val
-            for k in range(N-1):
-                if (((p[k, j, i] < p0) and (p[k+1, j, i] > p0)) or
-                   ((p[k, j, i] > p0) and (p[k+1, j, i] < p0))):
-                    dp = p[k+1, j, i] - p[k, j, i]
-                    dp0 = p0 - p[k, j, i]
-                    dq = q[k+1, j, i] - q[k, j, i]
-                    q_iso[j, i] = q[k, j, i] + dq*dp0/dp
+    with nogil:
```

Comment:
Cython docs says that `nogil` is fair game with memory views.  Not sure if that is "safe."  I do know that the cython compiler tries to be smart and tells you when a problem might occurs.  For example, I eliminated the temporary variables (`dq`, `dp0`, and `dp`) because the compiler told me so.

(It used to be the other way around... When did we start receiving orders from compilers?)